src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -0,0 +1,1839 @@
+                new object[] { false, TestCaseType.Json400KB, SR.Json400KB},
+            };
+
+        // TestCaseType is only used to give the json strings a descriptive name within the unit tests.
+        public enum TestCaseType
+        {
+            HelloWorld,
+            Basic,
+            BasicLargeNum,
+            ProjectLockJson,
+            FullSchema1,
+            DeepTree,
+            BroadTree,
+            LotsOfNumbers,
+            LotsOfStrings,
+            Json400B,
+            Json4KB,
+            Json40KB,
+            Json400KB,
+        }
+
+        private static string ReadHelloWorld(JToken obj)
+        {
+            string message = (string)obj["message"];
+            return message;
+        }
+
+        private static string ReadJson400KB(JToken obj)
+        {
+            var sb = new StringBuilder(250000);
+            foreach (JToken token in obj)
+            {
+                sb.Append((string)token["_id"]);
+                sb.Append((int)token["index"]);
+                sb.Append((string)token["guid"]);
+                sb.Append((bool)token["isActive"]);
+                sb.Append((string)token["balance"]);
+                sb.Append((string)token["picture"]);
+                sb.Append((int)token["age"]);
+                sb.Append((string)token["eyeColor"]);
+                sb.Append((string)token["name"]);
+                sb.Append((string)token["gender"]);
+                sb.Append((string)token["company"]);
+                sb.Append((string)token["email"]);
+                sb.Append((string)token["phone"]);
+                sb.Append((string)token["address"]);
+                sb.Append((string)token["about"]);
+                sb.Append((string)token["registered"]);
+                sb.Append((double)token["latitude"]);
+                sb.Append((double)token["longitude"]);
+
+                JToken tags = token["tags"];
+                foreach (JToken tag in tags)
+                {
+                    sb.Append((string)tag);
+                }
+                JToken friends = token["friends"];
+                foreach (JToken friend in friends)
+                {
+                    sb.Append((int)friend["id"]);
+                    sb.Append((string)friend["name"]);
+                }
+                sb.Append((string)token["greeting"]);
+                sb.Append((string)token["favoriteFruit"]);
+
+            }
+            return sb.ToString();
+        }
+
+        private static string ReadHelloWorld(JsonElement obj)
+        {
+            string message = obj.GetProperty("message").GetString();
+            return message;
+        }
+
+        private static string ReadJson400KB(JsonElement obj)
+        {
+            var sb = new StringBuilder(250000);
+
+            foreach (JsonElement element in obj.EnumerateArray())
+            {
+                sb.Append(element.GetProperty("_id").GetString());
```

Comment:
This is so much more verbose. Maybe we should reconsider adding the explicit casts and the indexer like property name  search.
```C#
sb.Append((string)element["_id"]);
```